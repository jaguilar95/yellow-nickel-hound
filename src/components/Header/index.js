import React from "react";
import Nav from "../Nav";

function Header(props) {
  const { sections = [] } = props;

  return (
    <header className="flex-row px-1">
      <h2>
        <a href="/">j.aguilar</a>
      </h2>
      <Nav sections={sections} />
    </header>
  );
}

export default Header;
