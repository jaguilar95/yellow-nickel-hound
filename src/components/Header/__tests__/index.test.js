import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Header from "..";

const sections = [
  { name: "portraits", description: "Portraits of people in my life" },
];
const currentSection = { name: "About Me" };
const mockSetCurrentSection = jest.fn();

afterEach(cleanup);

describe("Header component", () => {
  it("renders", () => {
    render(
      <Header
        sections={sections}
        currentSection={currentSection}
        setCurrentSection={mockSetCurrentSection}
      />
    );
  });

  it("header render matches snapshot", () => {
    const { asFragment } = render(
      <Header
        sections={sections}
        currentSection={currentSection}
        setCurrentSection={mockSetCurrentSection}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
