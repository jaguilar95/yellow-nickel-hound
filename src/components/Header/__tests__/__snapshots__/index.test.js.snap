// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component header render matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="flex-row px-1"
  >
    <h2>
      <a
        href="/"
      >
        j.aguilar
      </a>
    </h2>
    <div>
      <nav>
        <ul
          class="flex-row"
        >
          <li
            class="mx-2 false"
          >
            <span>
              portraits
            </span>
          </li>
        </ul>
      </nav>
    </div>
  </header>
</DocumentFragment>
`;
